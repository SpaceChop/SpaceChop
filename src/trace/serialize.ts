// Serialize anything passed to the log.
const serialize = (message: any) => {
  if (message === undefined) {
    return 'undefined';
  }
  if (message === Infinity) {
    return 'Infinity';
  }
  if (typeof message === 'string') {
    if (message.length === 0) {
      return '';
    }
    return message;
  }
  return JSON.stringify(message);
};

export default serialize;
