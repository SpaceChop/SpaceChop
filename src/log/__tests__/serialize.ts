import serialize from '../serialize';

describe('Serialize', () => {
  describe('should handle object', () => {
    it('null', () => {
      const obj = null;
      const out = serialize(obj);
      expect(typeof out).toBe('string');
      expect(out).toMatchSnapshot();
    });

    it('empty', () => {
      const obj = {};
      const out = serialize(obj);
      expect(typeof out).toBe('string');
      expect(out).toMatchSnapshot();
    });

    it('default', () => {
      const obj = {
        a: 1,
        b: 2,
      };
      const out = serialize(obj);
      expect(typeof out).toBe('string');
      expect(out).toMatchSnapshot();
    });

    it('nested', () => {
      const obj = {
        a: 1,
        b: 2,
        c: {
          a: 1,
          b: 1,
        },
      };
      const out = serialize(obj);
      expect(typeof out).toBe('string');
      expect(out).toMatchSnapshot();
    });
  });

  describe('should handle string', () => {
    it('empty', () => {
      const str = '';
      const out = serialize(str);
      expect(typeof out).toBe('string');
      expect(out).toMatchSnapshot();
    });

    it('default', () => {
      const str = 'hello world';
      const out = serialize(str);
      expect(typeof out).toBe('string');
      expect(out).toMatchSnapshot();
    });
  });

  describe('should handle number', () => {
    it('NaN', () => {
      const num = NaN;
      const out = serialize(num);
      expect(typeof out).toBe('string');
      expect(out).toMatchSnapshot();
    });

    it('undefined', () => {
      const num = undefined;
      const out = serialize(num);
      expect(typeof out).toBe('string');
      expect(out).toMatchSnapshot();
    });

    it('default', () => {
      const num = 1;
      const out = serialize(num);
      expect(typeof out).toBe('string');
      expect(out).toMatchSnapshot();
    });

    it('negative', () => {
      const num = -1;
      const out = serialize(num);
      expect(typeof out).toBe('string');
      expect(out).toMatchSnapshot();
    });

    it('large', () => {
      const num = 10 ** 10000;
      const out = serialize(num);
      expect(typeof out).toBe('string');
      expect(out).toMatchSnapshot();
    });
  });

  describe('should handle array', () => {
    it('empty', () => {
      const arr = [];
      const out = serialize(arr);
      expect(typeof out).toBe('string');
      expect(out).toMatchSnapshot();
    });

    it('default', () => {
      const arr = [1, 'hej', -1, NaN];
      const out = serialize(arr);
      expect(typeof out).toBe('string');
      expect(out).toMatchSnapshot();
    });
  });
});
