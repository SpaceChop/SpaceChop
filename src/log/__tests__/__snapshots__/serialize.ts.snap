// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Serialize should handle array default 1`] = `
"1
hej
-1
null"
`;

exports[`Serialize should handle array empty 1`] = `""`;

exports[`Serialize should handle number NaN 1`] = `"null"`;

exports[`Serialize should handle number default 1`] = `"1"`;

exports[`Serialize should handle number large 1`] = `"Infinity"`;

exports[`Serialize should handle number negative 1`] = `"-1"`;

exports[`Serialize should handle number undefined 1`] = `"undefined"`;

exports[`Serialize should handle object default 1`] = `
"{
  \\"a\\": 1,
  \\"b\\": 2
}"
`;

exports[`Serialize should handle object empty 1`] = `"{}"`;

exports[`Serialize should handle object nested 1`] = `
"{
  \\"a\\": 1,
  \\"b\\": 2,
  \\"c\\": {
    \\"a\\": 1,
    \\"b\\": 1
  }
}"
`;

exports[`Serialize should handle object null 1`] = `"null"`;

exports[`Serialize should handle string default 1`] = `"hello world"`;

exports[`Serialize should handle string empty 1`] = `""`;
